pteranodon/plugins/ext_plugins/power/power.py

```diff
@@ -0,0 +1,128 @@
+import time
+from asyncio import AbstractEventLoop
+from logging import Logger
+from typing import Dict, Optional
+from collections import deque
+from subprocess import SubprocessError
+
+import numpy as np
+from mavsdk import System, telemetry
+
+from pteranodon.plugins.ext_plugins.abstract_custom_plugin import AbstractCustomPlugin
+from .tegrastats import Tegrastats
+
+
+class Power(AbstractCustomPlugin):
+    """
+    Generates battery usage information
+    """
+
+    def __init__(
+        self,
+        system: System,
+        loop: AbstractEventLoop,
+        logger: Logger,
+        base_plugins: Dict,
+        ext_args: Dict,
+    ) -> None:
+        super().__init__("power", system, loop, logger, base_plugins, ext_args)
+        self._telemetry = self._base_plugins["telemetry"]
+        self._param = self._base_plugins["param"]
+        self._capacity = self._param.get_param_float("BAT1_CAPACITY")
+        self._num_cells = self._param.get_param_int("BAT1_N_CELLS")
+        if self._ext_args["power"] is not None:
+            self._window_size = self._ext_args["power"][0]
+            try:
+                self._tegrastats = Tegrastats(self._ext_args["power"][1])
+                self._instantiated = True
+            except SubprocessError:
+                self._instantiated = False
+        else:
+            self._window_size = 10
+            try:
+                self._tegrastats = Tegrastats()
+                self._instantiated = True
+            except SubprocessError:
+                self._instantiated = False
+        self._window: deque = deque(maxlen=self._window_size)
+
+        @self._telemetry.register_battery_handler
+        def battery_handler(battery: telemetry.Battery):
+            self._window.append((battery, time.time()))
+
+    def get_instantaneous_wattage(self) -> Optional[float]:
+        """
+        Get the Current*Voltage value, or instantaneous wattage
+        :return: float : Instantaneous battery usage, Current*Voltage, None if battery has not been polled yet
+        """
+        if len(self._window) > 0:
+            batt_info = self._window[-1]
+            return self._param.get_param_float("BAT1_A_PER_V") * (
+                batt_info.voltage_v**2
+            )
+        return None
+
+    def get_hardware_wattage(self) -> Optional[float]:
```

Comment:
This should be changed to check if tegra got intialized, if not return None
This will allow for future hardware implementations